### PR TITLE
add tests for aggregation

### PR DESCRIPTION
**Pull request by kenjinp**
# NEW
Adds a simple query method to the ElasticSearchView intended to be used for aggregations

## Caveat
This query really could be used for everything, it might make sense to have a naked query response method in the future, but for now i'm just assuming it'll be used for aggregations, because they're quite uniform in our current use cases

### CODE REVIEW

File: nexus-sdk-js/src/__mocks__/helpers.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import { ViewsListResponse } from '../views';
 import {
   ElasticSearchViewResponse,
   ElasticSearchViewQueryResponse,
+  ElasticSearchViewAggrigationResponse,
```
Comment (mdupont) [inline nexus-sdk-js/src/__mocks__/helpers.ts]: aggr**i**gation? To rename I guess

File: nexus-sdk-js/src/views/ElasticSearchView/index.ts [hunk 2]
```diff
@@ -56,6 +56,21 @@ export interface ElasticSearchViewQueryResponse {
   took: number;
 }
 
+export interface Bucket {
+  key: string;
+  doc_count: number;
+}
+
+export interface ElasticSearchViewAggrigationResponse {
```
Comment (mdupont) [inline nexus-sdk-js/src/views/ElasticSearchView/index.ts]: Typo source is here

File: nexus-sdk-js/src/views/ElasticSearchView/index.ts [hunk 3]
```diff
@@ -56,6 +56,21 @@ export interface ElasticSearchViewQueryResponse {
...
+  aggregations: {
+    [key: string]: {
+      doc_count_error_upper_bound: number;
+      sum_other_doc_count: number;
+      buckets: Bucket[];
+    };
+  };
+}
+
 class ElasticSeachView {
```
Comment (mdupont) [inline nexus-sdk-js/src/views/ElasticSearchView/index.ts]: This one also has a typo from an earlier commit, why not fix it at the same time? :-D

File: nexus-sdk-js/src/views/ElasticSearchView/__tests__/ElasticSearchView.spec.ts [hunk 4]
```diff
@@ -55,7 +56,7 @@ describe('ElasticSearchView class', () => {
   });
 
   // Skipping for now as it throw unhandled promise rejection errors.
-  describe.skip('query()', () => {
+  describe('query()', () => {
     const mockHttpPost = <jest.Mock<typeof httpPost>>httpPost;
```
Comment (julienmachon) [inline nexus-sdk-js/src/views/ElasticSearchView/__tests__/ElasticSearchView.spec.ts]: I'm not sure how this works exactly. We have `jest-fetch-mock` for mocking the fetch API. You can find an example here https://github.com/BlueBrain/nexus-webapp-commons/blob/master/nexus-sdk-js/src/Project/__tests__/Project.spec.ts#L157-L160

File: nexus-sdk-js/src/views/ElasticSearchView/index.ts [hunk 5]
```diff
@@ -177,6 +192,33 @@ class ElasticSeachView {
       throw new ViewQueryError(error.message);
     }
   }
+
+  /**
+   * Query aggregations in elastic search
+   * using an elastic search Aggregation Query (as a JS Object) as input
+   * https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-terms-aggregation.html
+   *
+   * @param {Object} elasticSearchAggregationQuery
```
Comment (mdupont) [inline nexus-sdk-js/src/views/ElasticSearchView/index.ts]: Type should start with a capital E

File: nexus-sdk-js/src/views/ElasticSearchView/index.ts [hunk 6]
```diff
@@ -177,6 +192,33 @@ class ElasticSeachView {
...
+   * @returns {Promise<ElasticSearchViewAggrigationResponse>}
+   * @memberof ElasticSeachView
```
Comment (mdupont) [inline nexus-sdk-js/src/views/ElasticSearchView/index.ts]: Oh so many renames after fixing the original typo

Comment (julienmachon) [reply nexus-sdk-js/src/views/ElasticSearchView/index.ts]: F2 is your friend

File: nexus-sdk-js/src/views/ElasticSearchView/__tests__/ElasticSearchView.spec.ts (hunk 4 above)
Comment (kenjinp) [reply nexus-sdk-js/src/views/ElasticSearchView/__tests__/ElasticSearchView.spec.ts]: I've updated to use jest-fetch-mock :)

**Pull request merged**